python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -0,0 +1,921 @@
+        http://lsst-web.ncsa.illinois.edu/doxygen/x_masterDoxyDoc/_background_example.html
+        """
+        from . import utils as ipUtils
+
+        fp = source.getFootprint()
+        bbox = fp.getBBox()
+        bbox.grow(3)
+        posImg = afwImage.ImageF(posImage.getMaskedImage().getImage(), bbox, afwImage.PARENT)
+
+        # This code constructs the footprint image so that we can identify the pixels that are
+        # outside the footprint (but within the bounding box). These are the pixels used for
+        # fitting the background.
+        posHfp = afwDet.HeavyFootprintF(fp, posImage.getMaskedImage())
+        posFpImg = ipUtils.getHeavyFootprintSubimage(posHfp, grow=3)
+
+        isBg = np.isnan(posFpImg.getArray()).ravel()
+
+        data = posImg.getArray().ravel()
+        data = data[isBg]
+        B = data
+
+        x, y = np.mgrid[bbox.getBeginY():bbox.getEndY(), bbox.getBeginX():bbox.getEndX()]
+        x = x.astype(np.float64).ravel()
+        x -= np.mean(x)
+        x = x[isBg]
+        y = y.astype(np.float64).ravel()
+        y -= np.mean(y)
+        y = y[isBg]
+        b = np.ones_like(x, dtype=np.float64)
+
+        M = np.vstack([b]).T  # order = 0
+        if order == 1:
+            M = np.vstack([b, x, y]).T
+        elif order == 2:
+            M = np.vstack([b, x, y, x**2., y**2., x*y]).T
+
+        pars = np.linalg.lstsq(M, B)[0]
+        return pars
+
+    def genStarModel(self, bbox, psf, xcen, ycen, flux):
+        """Generate model (2-d array) of a 'star' (single PSF) centered at given coordinates
+
+        @param bbox Bounding box marking pixel coordinates for generated model
+        @param psf Psf model used to generate the 'star'
+        @param xcen Desired x-centroid of the 'star'
+        @param ycen Desired y-centroid of the 'star'
+        @param flux Desired flux of the 'star'
+
+        @return 2-d numpy.array of width/height matching input bbox,
+        containing PSF with given centroid and flux
+        """
+
+        # Generate the psf image, normalize to flux
+        psf_img = psf.computeImage(afwGeom.Point2D(xcen, ycen)).convertF()
+        psf_img_sum = np.nansum(psf_img.getArray())
+        psf_img *= (flux/psf_img_sum)
+
+        # Clip the PSF image bounding box to fall within the footprint bounding box
+        psf_box = psf_img.getBBox()
+        psf_box.clip(bbox)
+        psf_img = afwImage.ImageF(psf_img, psf_box, afwImage.PARENT)
+
+        # Then actually crop the psf image.
+        # Usually not necessary, but if the dipole is near the edge of the image...
+        # Would be nice if we could compare original pos_box with clipped pos_box and
+        #     see if it actually was clipped.
+        p_Im = afwImage.ImageF(bbox)
+        tmpSubim = afwImage.ImageF(p_Im, psf_box, afwImage.PARENT)
+        tmpSubim += psf_img
+
+        return p_Im
+
+    @staticmethod
+    def genDipoleModel(x, flux, xcenPos, ycenPos, xcenNeg, ycenNeg, fluxNeg=None,
```

Comment:
I have updated it so that it uses an internal method to call `aDipoleModel.makeModel`. This at least eliminates the need for the static method in `DipoleModel`.